src/php/Printer/TypePrinter/ArrayPrinter.php

```diff
@@ -4,13 +4,65 @@
 
 namespace Phel\Printer\TypePrinter;
 
+/**
+ * @implements TypePrinterInterface<array>
+ */
 final class ArrayPrinter implements TypePrinterInterface
 {
+    use WithColorTrait;
+
     /**
-     * @param mixed $form
+     * @param array $form
      */
     public function print($form): string
     {
-        return '<PHP-Array>';
+        $arr = $this->isList($form)
+            ? $this->formatValuesFromList($form)
+            : $this->formatKeyValuesFromDict($form);
+
+        return sprintf('<PHP-Array [%s]>', $this->color(implode(', ', $arr)));
+    }
+
+    private function isList(array $form): bool
+    {
+        return array_keys($form) === range(0, count($form) - 1);
+    }
+
+    private function formatValuesFromList(array $form): array
+    {
+        return array_map(
+            fn ($v) => $this->formatValue($v),
+            $form
+        );
+    }
+
+    private function formatKeyValuesFromDict(array $form): array
+    {
+        return array_map(
+            fn ($k, $v) => sprintf('%s:%s', $this->formatValue($k), $this->formatValue($v)),
+            array_keys($form),
+            $form
+        );
+    }
+
+    /**
+     * @param mixed $v
+     */
+    private function formatValue($v): string
+    {
+        if (is_string($v)) {
+            return sprintf('"%s"', $v);
+        }
+
+        return (string)$v;
```

Comment:
* If `$v` is `true` it will print `1` and not `true`.
* If `$v` is `null` it will print an empty string and not `null`.
* What is a PHP Array contains a Phel Array. Then this is not printed correctly.

I think this method must be recursively call the main Printer?

Reply:
Good point. I just changed 👌🏻